backend/dataall/modules/metadata_forms/db/metadata_form_models.py

```diff
@@ -0,0 +1,149 @@
+
+
+class MetadataForm(Base):
+    __tablename__ = 'metadata_form'
+    uri = Column(String, primary_key=True, default=utils.uuid('metadata_form'))
+    name = Column(String, nullable=False)
+    description = Column(String, nullable=False)
+    SamlGroupName = Column(String, nullable=False)
+    visibility = Column(String, nullable=False)  # enum MetadataFormVisibility
+    homeEntity = Column(String, nullable=True)
+
+
+class MetadataFormEnforcementRule(Base):
+    __tablename__ = 'metadata_form_enforcement_rule'
+    uri = Column(String, primary_key=True, default=utils.uuid('rule'))
+    metadataFormUri = Column(String, ForeignKey('metadata_form.uri'))
+    level = Column(String, nullable=False)  # enum MetadataFormEnforcementScope
+    entityTypes = Column(ARRAY(String), nullable=False)  # enum MetadataFormEntityTypes
+    severity = Column(String, nullable=False)  # enum MetadataFormEnforcementSeverity
+
+
+class MetadataFormField(Base):
+    __tablename__ = 'metadata_form_field'
+    metadataFormUri = Column(String, ForeignKey('metadata_form.uri'))
+    uri = Column(String, primary_key=True, default=utils.uuid('field'))
+    name = Column(String, nullable=False)
+    type = Column(String, nullable=False)  # enum MetadataFormFieldType
+    required = Column(Boolean, nullable=False)
+    glossaryTerm = Column(String, ForeignKey('term_link.linkUri'), nullable=True)
+    possibleValues = Column(ARRAY(String), nullable=True)
+
+
+class AttachedMetadataForm(Base):
+    __tablename__ = 'attached_metadata_form'
+    metadataFormUri = Column(String, ForeignKey('metadata_form.uri'), nullable=False)
+    uri = Column(String, primary_key=True, default=utils.uuid('attached_form'))
+    entityUri = Column(String, nullable=False)
+    entityType = Column(String, nullable=False)
+
+
+class AttachedMetadataFormField(Base):
+    __tablename__ = 'attached_metadata_form_field'
+    attachedFormUri = Column(String, ForeignKey('attached_metadata_form.uri'), primary_key=True)
+    fieldUri = Column(String, ForeignKey('metadata_form_field.uri'), primary_key=True)
+    type = Column(String, nullable=False)
+    field = relationship('MetadataFormField', backref='attached_fields')
+
+    __table_args__ = (PrimaryKeyConstraint('attachedFormUri', 'fieldUri'),)
+    __mapper_args__ = {'polymorphic_identity': 'attached_metadata_form_field', 'polymorphic_on': type}
+
+    @property
+    def value(self):
+        raise NotImplementedError('Basic AttachedMetadataFormField has no implemented property value')
+
+    @validates('type')
+    def update_type(self, key, new_type):
+        if new_type != self.field.type:
+            raise ValueError("Value type doesn't match field type")
+
+
+class StringAttachedMetadataFormField(AttachedMetadataFormField):
+    __tablename__ = 'string_attached_metadata_form_field'
+    attachedFormUri = Column(String, primary_key=True)
+    fieldUri = Column(String, primary_key=True)
+    sValue = Column(String, nullable=False)
```

Comment:
why sValue/bValue/iValue/gtValue and not just value for the column name?

Reply:
Hm. It was an artifact from when it was one table. Totally forgot to redo

Reply:
[f961805](https://github.com/data-dot-all/dataall/pull/1422/commits/f961805854f500e7dceaac87790245df73fdc586)